eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```
@@ -232,14 +233,35 @@ jobs:
       Location: ${{ parameters.Location }}
       SubscriptionConfiguration: ${{ parameters.SubscriptionConfiguration }}
 
-  - script: ctest -V -C Debug --tests-regex ${{ parameters.CtestRegex }} --no-compress-output -T Test
+    # For non multi-config generator use the same build configuration to run tests
+    # We don't need to set it to invoke ctest 
+  - script: ctest -V --tests-regex ${{ parameters.CtestRegex }} --no-compress-output -T Test
     workingDirectory: build
     displayName: ctest
     # Runs only if test-resources are happly deployed.
     # unit-tests runs for those configs where samples are not ran.
     # This enables to run tests and samples at the same time as different matrix configuration.
     # Then unit-tests runs, samples should not run.
-    condition: and(succeeded(), ne(variables['RunSamples'], '1'))
+    condition: and(
+      succeeded(),
+      ne(variables['RunSamples'], '1'),
+      not(contains(variables['OSVmImage'], 'MMS'))
```

This check will not work for you. I think you want something like:
```suggestion
      eq(variables['OSName'], 'windows')
```

Note that we set OSName based on image name in our verify-os steps https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/scripts/Verify-AgentOS.ps1.